tests/common/setup.py

```diff
@@ -22,16 +22,19 @@
 from tempfile import mkdtemp
 
 from hypothesis import settings, unlimited
+from hypothesis.errors import HypothesisDeprecationWarning
 from hypothesis.configuration import set_hypothesis_home_dir
 from hypothesis.internal.charmap import charmap, charmap_file
 from hypothesis.internal.coverage import IN_COVERAGE_TESTS
 
 
 def run(deprecations_as_errors=True):
     warnings.filterwarnings('error', category=UnicodeWarning)
-    # This catches deprecations in our dependencies, as well as internally
-    # (because HypothesisDeprecationWarning subclasses DeprecationWarning)
-    if deprecations_as_errors:  # disabled for old versions of Django
+    warnings.filterwarnings('error', category=HypothesisDeprecationWarning)
+    # We further enable deprecations as errors for all dependencies,
+    # unless disabled for Django 1.8, or by downstream (see issue #969)
+    disable_werror = os.environ.get('HYPOTHESIS_TESTS_NO_WERROR') == 'true'
```

Comment:
Two questions:
0. Shouldn't we document this?
1. Should we provide pytest cli option to control this?
2. Should we check env variable value our just ensure that this variable is set? For instance, standard Python env variables works in this way, so there is no difference between PYTHONOPTIMIZE=1 and PYTHONOPTIMIZE=true and PYTHONOPTIMIZE=yes-please.

Reply:
That’s three questions. 😉 

I’d prefer a pytest CLI option – particularly because tox doesn’t pass through environment variables by default, so at minimum this needs a change to `tox.ini` or I’m pretty sure won’t work. A CLI option is more explicit in that regard.

Reply:
Eh, I was hope I'll fix my misunderstood before anyone noticed. Now it looks awkward (:

Reply:
Hey, I liked your questions!

Reply:
Get all back, no worries (: